### PR TITLE
Upgrade Semantic Kernel to 1.0.0-beta4

### PR DESCRIPTION
**Pull request by ciprianjichici**
Upgrade the version of Semantic Kernel to 1.0.0-beta4 and implement several changes resulting from changes introduced in the new version.

### CODE REVIEW

File: VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        public TextSummaryPlugin(
+            string promptTemplate,
+            int maxTokens,
+            IKernel kernel)
+        {
+            _kernel = kernel;
+            _summarizeConversation = kernel.CreateSemanticFunction(
+                promptTemplate,
+                pluginName: nameof(TextSummaryPlugin),
+                description: "Given a text, summarize the text.",
```
Comment (markjbrown) [inline VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs:22]: Is the system prompt for this still coming from blob storage to drive the behavior?

File: VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                requestSettings: new OpenAIRequestSettings
+                {
+                    MaxTokens = maxTokens,
+                    Temperature = 0.1,
+                    TopP = 0.5
```
Comment (markjbrown) [inline VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs:27]: Did you test these settings? In other configurations I had this as determinstic as possible with temp=0 and top_p=1 to keep it from going off the rails. Not sure if these are ok. just haven't tried these values.

**Comment by markjbrown** [comment]
@ciprianjichici had a couple of comments/questions but I'm going to merge this. I REALLY love how the project changed. Some things in here just seem more clean now.

Thanks!!!

File: VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs (hunk 1 above)
Comment (ciprianjichici) [reply VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs:22]: Yes, still coming from the blob storage account.

File: VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs (hunk 2 above)
Comment (ciprianjichici) [reply VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs:27]: For summarization, I prefer a little bit of freedom, but will add a comment to point out that folks should try with temp=0 and top_p=1 as well.

**Pull request merged**